### PR TITLE
Service version const strings

### PR DESCRIPTION
**Pull request by Jinming-Hu**
closes https://github.com/Azure/azure-sdk-for-cpp/issues/1056

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Comment by Jinming-Hu** [comment]
@msftbot merge this PR after @katmsft approves

**Review by katmsft — APPROVED** [review]

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:135]: Shouldn't this be an enum/extensible enum?

In which case, should this be a class with `const static` values?
https://azure.github.io/azure-sdk/cpp_implementation.html#enumerations

cc @JeffreyRichter

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:135]: Extensible enum is used when client side needs to parse a value returned from server side. I don't see when this SDK or customer needs to do that. Actually I don't even know when this service version string will be used.

Comment (ahsonkhan) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:135]: I thought this version was going to be passed as the API version header to the service, but might be mistaken. @tg-msft, are we supposed to plumb the `ServiceVersion` string to any of the storage service API calls?

Comment (JeffreyRichter) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:135]: Jinming is correct; this should not be an extensible enum.
I think we just want a string field in the XxxClient options struct called ApiVersion. By default, we set this to the api-version that corresponds to the latest version of the service and this version is what we test & support.
We want to allow the customer to pass their own string if they want to throw caution to the wind and just try an arbitrary api-version. Usually, the customers experience will be fine if they do this but, we need to say that this is not a tested & supported scenario. This enables scenarios like Azure Stack Hub and other non-public clouds to work that do not have the latest version of the service available.

Comment (ahsonkhan) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:135]: Aside from the `ApiVersion` string field, why not help such a customer by providing them with the list of valid api version strings to choose from, rather than let them pick any arbitrary free-form string?

This way, let's say we ship storage v2, we will have V2020_02_10 and say V2021_12_31, and then the user can set `ApiVersion` to either one of those (still best effort), rather than the user having to fish out from storage service docs the version strings.

Comment (JeffreyRichter) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:135]: Because we don't support this. It feels odd to me to here are a bunch of possible values and we don't guarantee any of them will work for you.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:135]: Hi @JeffreyRichter , are you suggesting that I should remove what I've added in this PR and add an extra `std::string` field into `BlobClientOptions`, the default value for this `std::string` should be the latest API version we support, while we still allow our customers to change this `std::string` value if they want to try another API version?

Comment (JeffreyRichter) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:135]: Yes, exactly.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:135]: @JeffreyRichter Thanks for your clarification!

**Pull request merged**
